src/query/anonymization.c

```diff
@@ -622,6 +622,15 @@ static void compile_anonymizing_query(Query *query, List *personal_relations, An
 
 static bool is_anonymizing_query(Query *query, List *personal_relations)
 {
+  if (query->limitCount != NULL)
+  {
+    Node *limit = unwrap_cast(query->limitCount);
+    if (IsA(limit, Const) &&
+        ((Const *)limit)->consttype == INT4OID &&
+        DatumGetInt32(((Const *)limit)->constvalue) == 0)
+      return false; /* No need for anonymization if no rows are requested. */
```

Comment:
If we consider timing attacks, then one could run all sorts of measurements based on the response time.
The simplest that comes to mind is a conditional `sleep()` if a particular row exists.

Reply:
Good point! 
Although I can't figure out how would one do that, since `pg_sleep` returns `void`, so it can't be used in the `WHERE` clause and the `SELECT` clause won't run at all.
But to be sure, let's detect the always false `WHERE` clause case instead.

Reply:
Alternatively, we can allow either case here, but walk the final plan again to ensure that there are no table or index scans against protected relations (the planner must have discarded them). This way we know no sensitive data will be read.

Reply:
That sounds somewhat complicated...

Reply:
I see the following field in `PlannedStmt`: `List *relationOids; /* OIDs of relations the plan depends on */`

If we "bypass" anon with the above heuristics, we can make sure that `relationOids` is free of sensitive tables. Sounds good in theory but I have no idea if the planner does that simplification or not..

Reply:
Seems that relation Oids are always present in the `PlannedStmt` object. So either they are not simplified, or it occurs later in the pipeline. In any case, it doesn't look like something we can use.

Reply:
From my observation `LIMIT 0` preserves the table scanning whereas `WHERE 1 <> 1` or other dead conditions drop it. The latter should be considered safe.